components/camel-cdi/src/main/java/org/apache/camel/cdi/CamelContextProducer.java

```diff
@@ -64,7 +66,15 @@
 
     @Override
     public T produce(CreationalContext<T> ctx) {
-        T context = super.produce(ctx);
+        T context;
+
+        if (getDelegate() instanceof ContextCreateStrategyInjectionTarget) {
```

Comment:
The `if` statement reveals an asymmetrical design. Either `ContextCreateStrategyInjectionTarget` is composed everywhere, or it can be removed and the logic always applied in `CamelContextProducer`.